frontend/server/minio-helper.ts

```diff
@@ -39,22 +39,21 @@ export interface MinioClientOptionsWithOptionalSecrets extends Partial<MinioClie
 export async function createMinioClient(config: MinioClientOptionsWithOptionalSecrets) {
   if (!config.accessKey || !config.secretKey) {
     try {
-      if (await awsInstanceProfileCredentials.ok()) {
-        const credentials = await awsInstanceProfileCredentials.getCredentials();
-        if (credentials) {
-          const {
-            AccessKeyId: accessKey,
-            SecretAccessKey: secretKey,
-            Token: sessionToken,
-          } = credentials;
-          return new MinioClient({ ...config, accessKey, secretKey, sessionToken });
-        }
-        console.error('unable to get credentials from AWS metadata store.');
+      const credentials = fromNodeProviderChain();
```

Comment:
Is it possible to detect the source of object storage is AWS first, and perform IRSA only when it is AWS S3? (Pattern like "arn:aws:s3:::$mybucket/*") If it is generic S3 API, then we shouldn't use IRSA solution.

Reply:
@zijianjoy  The code that used to be in this block was checking for AWS metadata store credentials. The fromNodeProviderChain() will try to access IRSA credentials then fallback and one of those would be AWS metadata store credentials which retains the exsiting functionality. 

If the option is preferred, then what I can do is call the existing https://github.com/kubeflow/pipelines/blob/834d966033ebb8284981e86b831c2bb086ef30ba/frontend/server/aws-helper.ts#L54 and have this IRSA code live as an if statement above the rest of the code in this method. i.e (if isS3Endpoint) ->my new code

Reply:
Hello Ryan, thank you for replying to my comment, Sounds good on using `isS3Endpoint` , I would also suggest adding a comment to explain that which logic area is specific to AWS, and which logic is generic S3.

Reply:
@zijianjoy What is an example of generic S3 which has `s3://` in the source/URL?
https://github.com/kubeflow/pipelines/blob/master/frontend/server/handlers/artifacts.ts#L33